### PR TITLE
CLN: Remove last usages of the Any type

### PR DESCRIPTION
**Pull request by janbjorge**
Extends: https://github.com/equinor/fmu-dataio/pull/484

### CODE REVIEW

File: src/fmu/dataio/_objectdata_provider.py [hunk 1]
```diff
@@ -699,6 +704,9 @@ def _get_columns(self) -> list[str]:
             columns = list(self.obj.columns)
         else:
             logger.debug("arrow")
+            from pyarrow import Table
```
Comment (mferrera) [inline src/fmu/dataio/_objectdata_provider.py:707]: I wonder if and when these lazy imports start to become numerous enough to bite us later on... the import doesn't seem excessively slow to my mind, about half a second (at least as far as Python is concerned). Probably few enough to be OK now, but will there be more?

**Review by mferrera — APPROVED** [review]
This looks OK to me, but I think it may be best to have @jcrivenaes or @perolavsvendsen weigh in on it as well

File: src/fmu/dataio/_objectdata_provider.py (hunk 1 above)
Comment (janbjorge) [reply src/fmu/dataio/_objectdata_provider.py:707]: Ive wondered so as well. But i think for now, its fine. But if i go back and look at older versions it looks like there was a period of time where we could not really upon pyarrow being installed.

File: src/fmu/dataio/_filedata_provider.py [hunk 2]
```diff
@@ -176,7 +185,7 @@ def _get_path(self) -> tuple[Path, Path | None]:
 
     def _get_path_generic(
         self,
-        mode: Literal[FmuContext.REALIZATION, FmuContext.PREPROCESSED],
+        mode: FmuContext,
```
Comment (janbjorge) [inline src/fmu/dataio/_filedata_provider.py:188]: Discussed with @jcrivenaes ✅

**Pull request merged**